rest/main/main.go

```diff
@@ -218,3 +230,29 @@ func findAManagementIP() string {
 	glog.Warning("Could not find a management address!!")
 	return ""
 }
+
+// errorMsgPrefixes identifies the important error messages logged by
+// the standard http library that should not be missed
+var errorMsgPrefixes = [][]byte{
+	[]byte("http: Accept error"),
+	[]byte("http: panic"),
+	[]byte("http2: panic"),
+}
+
+// serverLog forwards the http library's logs to glog
+var serverLog = log.New(logWriter{}, "", 0)
+
+type logWriter struct{}
+
+func (logWriter) Write(p []byte) (int, error) {
+	for _, pfx := range errorMsgPrefixes {
```

Comment:
While it works this is a very brittle - it requires tracking changes of the http library error messages - if, say, `http: panic` changes to `http:  panic` the message will not be logged (and everybody will be happy that there is no more panics from this module).

Also a loop checking all patterns is not particularly scalable.

Reply:
> While it works this is a very brittle

Totally agree.. but I cannot think of any other better way of filtering. Intention here is to avoid flooding log due to a buggy clients (like "TLS handshake error" messages) but not loose important server side errors. I can think of only accept failures and handler panic cases. I checked the go sources from multiple releases and these message formats have not changed. Hopefully they will remain unchanged in future releases as well.

Another alternative I considered was to have panic recovery for our handlers and blindly suppress logs from http library (using `io.Discard` as the log writer). But we will loose accept errors and panics happening outside our handler.

Reply:
How about using the [regex](https://pkg.go.dev/regexp) library to define rules for logs to be printed out?

Reply:
I feel direct []byte search is faster and lighter than regex based search

Reply:
You might be right, but (1) finding out what is the difference (and if it matters) would require writing and running a benchmark test as this library is optimized to run efficiently and (2) using this approach would make the solution less brittle as it allows for searching for patterns instead of fixed prefixes.

Reply:
We did our analysis and our initial implementations had used regex and that was very slow and hence we came up with this approach and it proved us to be performant and hence we made this change elsewhere as well.

Reply:
OK